### PR TITLE
Calypso-build: Improve Jest setup

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

- Move `jest.config.js` to `jest-preset.js` so it can be used as a jest preset.
- Add jest babel transform to load calypso-build babel configuration.
- Reorganize jest transforms under `@automattic/calypso-build/jest/transform/`.

#### Testing
- Use `npm link` to test this in another project as a preset.
- Calypso tests should all pass.

### CODE REVIEW

**Comment by sirreal** [comment]
Enzyme isn't being set up in the preset. I'll investigate…

**Comment by sirreal** [comment]
> Enzyme isn't being set up in the preset. I'll investigate…

I think this is related to being run via `npm link` instead of as an installed package. I think this is working well.

**Review by ockham — APPROVED** [review]
This is great, Jon -- thanks a lot! Much more elegant with the preset :+1: 

Works fine for Calypso in my testing, and also for 28-gh-synthetic-pistachio (except for Enzyme, but :crossed_fingers: that it's really just due to `npm link`).

:shipit:

**Review by tyxla — APPROVED** [review]
Noiceeee 🚢

File: packages/calypso-build/package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
 		"@wordpress/browserslist-config": "2.3.0",
 		"@wordpress/dependency-extraction-webpack-plugin": "1.1.0",
 		"autoprefixer": "9.4.4",
+		"babel-jest": "^24.9.0",
```
Comment (simison) [inline packages/calypso-build/package.json]: Note that other versions in the package are pinned.

**Pull request merged**